File: solutions/lua/perfect-numbers/1/perfect-numbers.lua
local function aliquot_sum(n)
	local sum = 0
	local fac = 1
	while fac < n do
		if n % fac == 0 then
			sum = sum + fac
		end
		fac = fac + 1
	end
	return sum
end

local function classify(n)
	local sum = aliquot_sum(n)
	if sum < n then return "deficient" end
	if sum == n then return "perfect" end
	if sum > n then return "abundant" end
end

return { aliquot_sum = aliquot_sum, classify = classify }

File: solutions/ocaml/perfect-numbers/1/perfect_numbers.ml
let aliquot_sum n =
  let rec aux sum fac =
    if fac < n then
      aux (if (n mod fac) = 0 then sum + fac else sum) (fac + 1)
    else
      sum
  in aux 0 1

let classify (n: int): (string, string) result =
  if n <= 0 then
    Error "Classification is only possible for positive integers."
  else
  let sum = aliquot_sum n in
  if sum < n then Ok "deficient" else
  if sum == n then Ok "perfect" else
  Ok "abundant"
